saving notes and feedback from day 3 (#36)
coderefinery/2020-05-25-online: hackmd-day3.md

---

> permalink: /hackmd-day1/

Author: ```suggestion
permalink: /hackmd-day3/
```

### Centralized workflow exercise

https://coderefinery.github.io/git-collaborative/02-centralized/#centralized-workflow-exercise

* You go through most of this lesson, see if you can get to some discussion
* Each person should be able to make a pull request and have it merged.
* 20 minutes, ending around xx:15.  Break afterwards.

### git-collaborative continued

* A quick question, is the pull request on github similar to a git pull? because the name is confusing for a merge of branches vs a pull from a remote
    * The name comes from "I want the administrator to do a pull from me".  In reality, it's an internal merge within github.

* If you delete a local branch after merging, the remote branch still exists. Does this mean that there is a another step to delete the remote branch, and does it matter?
    * Yes, another step.
    * You can delete remote branches from your terminal with `git push origin --delete branchname`. What I often do is to go to GitHub and delete them there by clicking on a trashcan in the branches overview.

> * I deleted the feature branch on GitHub but it shows up on my computer when I do a “git branch -a”

Reviewer: ```suggestion
* I deleted the feature branch on GitHub but it shows up on my computer when I do a `git branch -a`
```

    * `git fetch --prune` should remove these.
    * Since git is decentralized, these views don't automatically update.

> * How do synchronise the outputs of "git branch -a” across all collaborators?

Reviewer: ```suggestion
* How do synchronise the outputs of `git branch -a` across all collaborators?
```

        * yes - assuming you originally cloned from your own copy
        * `git remote -v` should clarify what is what
        * when I do `git remote -v`, it says (fetch) and (push) at the end. What does it mean? And it has it for both upstream and origin
        * There's a way to have separate paths for pushing and pulling.  I hardly ever use this, but could be useful in some cases.
        * The important part in the `git remote -v` is the web address, check whether origin refers to your repository where you can write to.
        * If you accidentally cloned the "wrong" one, there is no need to delete and re-clone, you can modify origin (`git remote set-url origin otheraddress`) or to add your fork as an additional remote (`git remote add myfork myforkaddress`) and refer to `myfork` instead of `origin`

* Is flow: central->fork->clone usually then updated central->clone->fork path (ie via local work directory to fork)
  - yes we will prefer to clone the fork, since this is the repo you can write to
  - and we update forks typically via the local repository (I wish there was a more automatic way and always wondered why GitHub does not have simply a button to update forks)

> - 'git push origin "my-new-fork"' results in "remote: Invalid username or password"?

Reviewer: ```suggestion
- `git push origin "my-new-fork"` results in "remote: Invalid username or password"?
```

>   - check what origin refers to with "git remote -v", maybe you are trying to push to a repository which is not yours? Also the syntax is `git push whereto whichbranch`

Reviewer: ```suggestion
  - check what origin refers to with `git remote -v`, maybe you are trying to push to a repository which is not yours? Also the syntax is `git push whereto whichbranch`
```

      - Rebasing is advanced and it modifies history and it is OK to not use it.
      - It can help creating more linear histories.
      - Ok, got it. Thanks!

- how to add a colaborator in a project on github? Is it under setting/action permission/add runner?
  - settings -> "manage access"

### Day 3 feedback

* Pro: The breakoutroom worked much better today. Examples are very clear and gave us good experience of working collaboratively.
* Cons: The only thing is to keep the  text in one place to not confuse people with what the text is. Try to be consistent in time structure for class (when breaks are).

* For the breakout rooms we managed to do very little, and that little was not even always correct. Next time that will have to be better prepared and at least tested, to make sure the instructions are up to date.

* The examples with central->distributed were very clear, to improve: some were missunderstanding "examples" as lines that should have been executed. The remote upstream namely would benefit from different backround for "do this", and "read this". But I liked a lot.

> * The first part of the main session was good, although really fast and I am not sure those not familiadayr were all able understand.

Reviewer: ```suggestion
* The first part of the main session was good, although really fast and I am not sure those not familiar were all able understand.
```

* The second part was really confusing and difficult to follow (Richard even intervened to ask what that was about) especially when the trainer went back & forth in the teaching material or did not share his terminal while typing.

* A thorough recap next Tuesday at the start of the session about what we did today will be highly appreciated.

- The main reason I registered for this workshop was to know more about "Collaborative distributed version control" and learn the best practices through simple exercises. I get out of here really frustrated.

>   - [name=Radovan] sorry to read this. I would love to hear what I/we can do to avoid this in future. Was it the presentation? The material? The exercises/examples? The timing? Speed?

Reviewer: ```suggestion
  - [name=Radovan] sorry to read this. I would love to hear what I/we can do to avoid this in future. Was it the presentation? The material? The exercises/examples? The timing? Speed?
  - Please contact us to let us know why.  In particular, I'm wondering - was it too fast or too slow, or a different topic than expected?  Something else?
```

- Feedback:
    - Pros:
        - Very well designed exercises
        - Clear and easy to follow explanations
    - Cons:
        - Time allocated to exercises a bit short
    - Overall: a big big up for the effort, the material and the energy put in this. Thank you!

* Maybe next time we could show how to do it firstly, then we redo it in our group. I think it is easy for following. This morning, I did not get most of lessons.

> - In our breakout rooms one person litterally monopolized all attention of the helper, systematically. That slowed down everything and significantly affected our sessions. It may be better, perhaps, to put absolute beginners together into a separate group and adapt the the training for them.

Reviewer: ```suggestion
- In our breakout rooms one person literally monopolized all attention of the helper, systematically. That slowed down everything and significantly affected our sessions. It may be better, perhaps, to put absolute beginners together into a separate group and adapt the the training for them.
  - We will try to make things are more balanced in the future.  It's quite hard to learn to say no to someone, so this is always a concern.
```
